core/src/test/scala/unit/kafka/server/AbstractFetcherManagerTest.scala

```diff
@@ -210,4 +217,103 @@ class AbstractFetcherManagerTest {
     verify(fetcher).maybeUpdateTopicIds(Set(tp1), topicIds)
     verify(fetcher).maybeUpdateTopicIds(Set(tp2), topicIds)
   }
+
+  @Test
+  def testExpandThreadPool(): Unit = {
+    val topicPartitions = mockTopicPartition(10, 100)
+    val fetcherManager = new AbstractFetcherManager[AbstractFetcherThread]("fetcher-manager", "fetcher-manager", 10) {
+      override def createFetcherThread(fetcherId: Int, sourceBroker: BrokerEndPoint): AbstractFetcherThread = {
+        new TestResizeFetcherThread(sourceBroker)
+      }
+    }
+    try {
+      resizeAndCheckFetcherPartitionDistribution(fetcherManager, topicPartitions, 25)
+    } finally {
+      fetcherManager.closeAllFetchers()
+    }
+  }
+
+  @Test
+  def testShrinkThreadPool(): Unit = {
+    val topicPartitions = mockTopicPartition(10, 100)
+    val fetcherManager = new AbstractFetcherManager[AbstractFetcherThread]("fetcher-manager", "fetcher-manager", 20) {
+      override def createFetcherThread(fetcherId: Int, sourceBroker: BrokerEndPoint): AbstractFetcherThread = {
+        new TestResizeFetcherThread(sourceBroker)
+      }
+    }
+    try {
+      resizeAndCheckFetcherPartitionDistribution(fetcherManager, topicPartitions, 5)
+    } finally {
+      fetcherManager.closeAllFetchers()
+    }
```

Comment:
`testShrinkThreadPool` is almost identical to `testExpandThreadPool`. Would it make sense to define `testResizeThreadPool(currentSize, newSize)` and use it in both?